src/react/await.tsx

```diff
@@ -0,0 +1,63 @@
+import { Await, AwaitProps } from "@remix-run/react";
+import {
+  suspendAll,
+  suspendAllSettled,
+  suspendAny,
+  suspendRace,
+} from "suspend-concurrently";
+
+type RenderProp<T> = React.ReactNode | ((value: T) => React.ReactNode);
+
+export interface AwaitAllProps<Resolve extends readonly unknown[] | []>
+  extends AwaitProps<Resolve> {
+  children: RenderProp<
+    Awaited<Promise<{ -readonly [P in keyof Resolve]: Awaited<Resolve[P]> }>>
```

Comment:
I think this can be simplified 🤔 Not sure how I ended up with the redundant nesting

```suggestion
    { -readonly [P in keyof Resolve]: Awaited<Resolve[P]> }
```